stp_zmq/zstack.py

```diff
@@ -753,17 +776,21 @@ def transmitThroughListener(self, msg, ident) -> Tuple[bool, Optional[str]]:
                          format(self, msg, ident))
             self.listener.send_multipart([ident, msg], flags=zmq.NOBLOCK)
         except zmq.Again:
-            return False, None
-        except InvalidMessageExceedingSizeException as ex:
-            err_str = '{}Cannot transmit message. Error {}'.format(CONNECTION_PREFIX, ex)
+            raise EAgainError(msg, ident, prefix=prefix)
+        except InvalidMessageExceedingSizeException as exc:
+            err_str = '{}Cannot transmit message. Error {}'.format(
+                CONNECTION_PREFIX, exc)
             logger.warning(err_str)
-            return False, err_str
-        except Exception as e:
-            err_str = '{}{} got error {} while sending through listener to {}'\
-                .format(CONNECTION_PREFIX, self, e, ident)
+            raise TooBigMessage(msg,
+                                ident=ident,
+                                msg_len=exc.actLen,
+                                max_len=exc.expLen,
+                                prefix=prefix) from exc
+        except Exception as exc:
+            err_str = ("{}{} got error {} while sending through listener to {}"
+                       .format(CONNECTION_PREFIX, self, exc, ident))
             logger.warning(err_str)
```

Comment:
Will we log the warning twice? Here and when catching the exceptions?

Reply:
Had decided to leave legacy logs firstly. Fixed for now.